internal/overlord/servstate/handlers.go

```diff
@@ -469,6 +469,11 @@ func (s *serviceData) exited(exitCode int) error {
 		switch action {
 		case plan.ActionIgnore:
 			logger.Noticef("Service %q %s action is %q, not doing anything further", s.config.Name, onType, action)
+			// On success we transition to state stopped.
+			if exitCode == 0 {
```

Comment:
We shoud have a success flag here so we're not doing the comparison in multiple places, and it becomes a bit less error prone and easier to refactor this logic in the future. For now it's only being done in two places nearby, so I'm going ahead regardless in the interest of unblocking you.